osu.Game/Scoring/ScoreManager.cs

```diff
@@ -88,6 +89,42 @@ protected override bool CheckLocalAvailability(ScoreInfo model, IQueryable<Score
             => base.CheckLocalAvailability(model, items)
                || (model.OnlineScoreID != null && items.Any(i => i.OnlineScoreID == model.OnlineScoreID));
 
+        protected override void PreImport(ScoreInfo model)
+        {
+            var latestFileInfos = new Dictionary<int, FileInfo>();
```

Comment:
what is this whole block doing?

Reply:
Some tracking issues were caused by `FileInfo` in `.Files` with same `ID` but different `ReferenceCount`, so it enumerates the list by desc and use latest entries to replace old entries (since the newly modified one must be ranked behind the previous one in `List<T>`).

Reply:
I'd personally be more fine with this sort of trick if we could ask *EF* for the newest entry, instead of essentially guessing at random that the entities at the end of the list are the newest entry.

Reply:
The modifications haven't been committed into database yet so there won't be latest entries from EF.

Reply:
very confused as to what committing to DB has to do with EF entry tracking. this is a `DbContext` level issue, correct? can't `DbContext.ChangeTracker.Entries()` be used or something?

Reply:
Both modified and unmodified entries are in `item`, so before calling `Add` there're no entries in ChangeTracker.